site-manage/site-manage-impl/impl/src/java/org/sakaiproject/sitemanage/impl/SiteManageServiceImpl.java

```diff
@@ -60,31 +74,37 @@
 @Slf4j
 public class SiteManageServiceImpl implements SiteManageService {
 
+	@Setter private AuthzGroupService authzGroupService;
     @Setter private ContentHostingService contentHostingService;
+    @Setter private CourseManagementService courseManagementService;
     @Setter private EntityManager entityManager;
     @Setter private EventTrackingService eventTrackingService;
     @Setter private LinkMigrationHelper linkMigrationHelper;
     @Setter private PreferencesService preferencesService;
     @Setter private SecurityService securityService;
     @Setter private ServerConfigurationService serverConfigurationService;
     @Setter private SessionManager sessionManager;
+    @Setter private ScoringService scoringService;
     @Setter private ShortenedUrlService shortenedUrlService;
     @Setter private SiteService siteService;
     @Setter private ToolManager toolManager;
     @Setter private TransactionTemplate transactionTemplate;
     @Setter private UserDirectoryService userDirectoryService;
     @Setter private UserNotificationProvider userNotificationProvider;
 
+    // Enough threads for Site imports and duplications
     @Setter private Integer siteImportThreadCount;
 
     private ExecutorService executorService;
     private Set<String> currentSiteImports;
+    private Set<String> currentSiteDuplicates;
 
     public void init() {
         // while this Set isn't cluster wide sessions are node specific
         // so this is only unsafe for more than one session performing an import on the same site
         // which is a really low percentage
         currentSiteImports = new ConcurrentSkipListSet<>();
+        currentSiteDuplicates = new ConcurrentSkipListSet<>();
```

Comment:
It probably makes sense for these to share the same data structure as you probably would never want someone scheduling an import or duplicate of the same site.

Reply:
okay addressed in latest commit